### ISSUE TITLE
Deferred label resolving only works for JMP/CALL

### ISSUE THREAD

**Issue by ranma** [vpelletier/dissn8#4]
For JUMP and CALL instructions, any unresolved label is remembered in label_referer_dict and later fixed up at the end.

However this would also be useful to have working for immediate args referring to labels, e.g. this fails:
```
B0MOV Y, #myromdata$M
B0MOV Z, #myromdata$L

myromdata DW 0x1234
```

with error:
```
Traceback (most recent call last):
[...]
  File "[...]dissn8/sn8/assn8.py", line 423, in p_identifier
    raise NameError(production[1])
NameError: myromdata
```

Unless the data is already know when then MOV op is assembled, then it works:
```
myromdata DW 0x1234

B0MOV Y, #myromdata$M
B0MOV Z, #myromdata$L
```

**Comment by vpelletier** [comment]
I agree.

This is going to take some effort, as I think this means all identifiers should be treated in this way. At a glance (I have not touched this code in a while), the grammar needs to be modified so rules which refer to identifiers (as opposed to rules declaring identifiers) tolerate almost them not being immediately resolvable (`@F` and `@B` being the most obvious exceptions, there may be others but they do not jump at me). Then, any grammar production which manipulates identifiers need to tolerate them not being resolved while still contributing to them (ex: bit selectors must affect the way the resolved identifier will affect produced machine code, and likewise for byte selectors). I did not walk all the way to machine code emission, but I think there may be something else needed, like how to shift/mask the value so it gets correctly mixed with the instruction once resolved.

At the moment I am busy on another project, and this change is resisting a bit too much to be done in parallel, so I'll have to come back to this later. If you are comfortable with python you could give it a jab, but unfortunately this code is not the cleanest I've written, so it may be unnecessarily difficult. The grammar looks overcomplex to me right now, but maybe I forgot the reasons which made it this way.
